integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/AppendDataDatasetBuilder.java

```diff
@@ -39,8 +40,16 @@ public List<OpenLineage.OutputDataset> apply(AppendData x) {
     LogicalPlan logicalPlan = (LogicalPlan) ((AppendData) x).table();
 
     if (logicalPlan instanceof DataSourceV2Relation) {
-      return new DataSourceV2RelationOutputDatasetBuilder(context, factory)
-          .apply((DataSourceV2Relation) logicalPlan);
+      DataSourceV2Relation relation = (DataSourceV2Relation) logicalPlan;
+      List<OpenLineage.OutputDataset> outputDatasets =
+          new DataSourceV2RelationOutputDatasetBuilder(context, factory)
+              .apply((DataSourceV2Relation) logicalPlan);
+
+      return ColumnLevelLineageUtils.buildColumnLineageDatasetFacet(context, relation.schema())
+          .filter(facet -> outputDatasets.stream().findFirst().isPresent())
+          .map(facet -> ColumnLevelLineageUtils.rewriteOutputDataset(outputDatasets.get(0), facet))
+          .map(el -> Collections.singletonList(el))
+          .orElse(outputDatasets);
```

Comment:
Wouldn't it make sense to move this logic to the `OpenLineageRunEventBuilder` so we don't have to include it in every OutputDataset builder? I think it's better to keep these kinds of internal APIs private so that extenders don't need to add these calls to every extension visitor they want to add - it'll make the visitor API difficult for others to maintain, as we may change the signatures or functionality going forward.

Reply:
Yeah, that's the great idea. Thanks for that.